src/app/codelabs/all-exercises/all-exercises.component.ts

```diff
@@ -0,0 +1,51 @@
+import { ExerciseBase } from './../../exercise/exercise/exercise.base';
+import { Component } from '@angular/core';
+import { Ng2TsExercises, ng2tsConfig } from '../../../../ng2ts/ng2ts';
+
+@Component({
+  selector: 'slides-all-exercises',
+  templateUrl: './all-exercises.component.html',
+  styleUrls: ['./all-exercises.component.css']
+})
+export class AllExercisesComponent {
+  private exercise;
+  private milestones;
+  public allFiles = {};
+
+  constructor(private exercises: Ng2TsExercises) {
+    this.exercise = exercises.getExercises(1, 1);
+    this.milestones = ng2tsConfig.milestones.map((milestone) => {
```

Comment:
Can you abstract this mapping to a private function and add a comment why/what the mapping achieves?

Something like:
```
this.milestones = ng2tsConfig.milestones.map(addIndexToAndFilterExercises);

...

/**
 * Adds the indexes to exercises on the milestone because _________.
 * Also filters exercises that don't have any files. 
 */
private addIndexToAndFilterExercises(milestone: Milestone) { ... 
```